### PR TITLE
Add login with Google using firebase

### PR DESCRIPTION
**Pull request by Ammar-Alesrawi**
This code is experimental, it has not been tested because I do not have a token at the moment
Two APIs were built: register + login.

### CODE REVIEW

File: Backend/src/lib/authentication/auth-user.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export class AuthUser {
   }
 
   async getFullUser(): Promise<User | undefined> {
-    if (this.fullUser || !this.isAuthorized()) {
+    if (this.fullUser ?? !this.isAuthorized()) {
       return this.fullUser;
     }
```
Comment (mahdy-nasr) [inline Backend/src/lib/authentication/auth-user.ts:34]:
```suggestion
    if ( !this.isAuthorized()) {
      return undefined;
    } else if (!!this.fullUser) { // we already cached it before.
      return this.fullUser;
    }
```
first:  The code original I wrote wasn't clear I think. This suggestion is more readable I think.
second: we are doing boolean check here, not assignment. we don't need `??`

File: Backend/src/lib/firebase.ts [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import admin from 'firebase-admin';
 import appConfig from '@/AppConfig/app.config';
-
```
Comment (mahdy-nasr) [inline Backend/src/lib/firebase.ts:3]: nit: always separate imports from the rest of the code with at least one empty line.

File: Backend/src/lib/authentication/auth-user.ts (hunk 1 above)
Comment (Ammar-Alesrawi) [reply Backend/src/lib/authentication/auth-user.ts:34]: We don't need to!! Here isn't it, one is enough :
 else if (!!this.fullUser)

Comment (mahdy-nasr) [reply Backend/src/lib/authentication/auth-user.ts:34]: didn't understand what you mean

Comment (Ammar-Alesrawi) [reply Backend/src/lib/authentication/auth-user.ts:34]: I mean the line should be like this:
else if (!this.fullUser)
not like that:
else if (!!this.fullUser)

**Review by mahdy-nasr — APPROVED** [review]

**Pull request merged**